crates/solver/src/settlement_simulation.rs

```diff
@@ -103,7 +104,7 @@ pub async fn simulate_and_error_with_tenderly_link(
                 // set a gas limit so we don't get failed simulations because of insufficient
                 // solver balance for the default ~15M gas limit. Limit to around the
                 // block gas limit / 2 (as settling bigger tx is unrealistic anyways).
-                .gas(15_000_000.into());
+                .gas(simulation_gas_limit.into());
```

Comment:
While reading your PR I noticed that the comment right above here says that the intended mainnet limit is 7.5m instead of 15m. Does anyone know what is right?
Either way I'd update this comment to remove the last sentence because the value is now passed in.

Reply:
30M is block gas limit while there is a gas target of 15M which is chased by increasing/decreasing block base fee.
Anyway, I would leave 15M as a default value for this parameter (as in this PR).